sdk/src/platform/Platform.ts

```diff
@@ -158,22 +163,54 @@ export default class Platform extends EventEmitter {
         return `${path + (path.includes('?') ? '&' : '?')}access_token=${(await this._auth.data()).access_token}`;
     }
 
-    public loginUrl({implicit, state, brandId, display, prompt, uiOptions, uiLocales, localeId}: LoginUrlOptions = {}) {
-        return this.createUrl(
-            `${this._authorizeEndpoint}?${qs.stringify({
-                response_type: implicit ? 'token' : 'code',
-                redirect_uri: this._redirectUri,
-                client_id: this._clientId,
-                state,
-                brand_id: brandId,
-                display,
-                prompt,
-                ui_options: uiOptions,
-                ui_locales: uiLocales,
-                localeId,
-            })}`,
-            {addServer: true},
-        );
+    public loginUrl({
+        implicit,
+        state,
+        brandId,
+        display,
+        prompt,
+        uiOptions,
+        uiLocales,
+        localeId,
+        usePKCE,
+    }: LoginUrlOptions = {}) {
+        let query: AuthorizationQuery = {
+            response_type: implicit ? 'token' : 'code',
+            redirect_uri: this._redirectUri,
+            client_id: this._clientId,
+            state,
+            brand_id: brandId,
+            display,
+            prompt,
+            ui_options: uiOptions,
+            ui_locales: uiLocales,
+            localeId,
+        };
+        this._codeVerifier = '';
+        if (usePKCE && !implicit) {
```

Comment:
Looks like these two flags are mutually exclusive, please throw an error if both are set. And add test for this too.